lib/addons/velero/index.ts

```diff
@@ -0,0 +1,239 @@
+        [key: string]: any;
+    };
+
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: VeleroAddOnProps = {
+    version: "2.23.6",
+    namespace: {
+        name: 'velero',
+        create: true
+    },
+    values:{
+        initContainers:[
+            {
+                name: "velero-plugin-for-aws",
+                image: "velero/velero-plugin-for-aws:v1.2.0",
+                imagePullPolicy: "IfNotPresent",
+                volumeMounts:[
+                    {
+                        mountPath: "/target",
+                        name: "plugins"
+                    }
+                ]
+            }
+        ],
+        configuration: {
+            provider: "aws",
+            backupStorageLocation:{
+                name: "default",
+                config:{}
+            },
+            volumeSnapshotLocation:{
+                name: "default",
+                config:{}
+            },
+        },
+        serviceAccount: {
+            server:{}
+        }
+    },
+
+};
+
+export class VeleroAddOn implements ClusterAddOn {
+
+    private options: VeleroAddOnProps;
+    constructor(props?: VeleroAddOnProps) {
+        if (props) {
+            // deepmerge the nested json files
+            this.options = deepmerge(defaultProps, props);
+        }
+        else {
+            this.options = defaultProps
+        }
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+        const props = this.options;
+        
+        let bucketName: string; // AWS S3 Bucketname
+        let veleroNamespace: string; // K8s namespace that Velero get deployed onto
+       
+        // Create S3 bucket if no existing bucket, create s3 bucket and corresponding KMS key
+        if ( !props.values.configuration.backupStorageLocation.bucket ){
+             console.log("existing S3 Bucket does not exists, creating S3 bucket");
+             const bucket = new s3.Bucket(cluster, 'velero-backup-bucket', {
+                encryption: s3.BucketEncryption.KMS_MANAGED, // Velero Known bug for support with S3 with SSE-KMS with CMK, thus it does not support S3 Bucket Key: https://github.com/vmware-tanzu/helm-charts/issues/83
+                blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL, // Block Public Access for S3
+                publicReadAccess: false
+            });
+
+            // S3 Bucket Policy for SSL Access
+            bucket.addToResourcePolicy(
+                new iam.PolicyStatement({
+                    sid: 'DenyHTTPTraffic',
+                    effect: iam.Effect.DENY,
+                    actions: ['s3:*'],
+                    resources: [bucket.arnForObjects('*')],
+                    principals: [new iam.AnyPrincipal()],
+                    conditions:{
+                        Bool:{
+                            'aws:SecureTransport': 'false',
+                        }
+                    }
+                })
+            );            
```

Comment:
I think this can be replaced with `enforceSSL: true` prop in the Bucket() creation.

Reply:
thanks. it works and resolved.